layers/+lang/ruby/funcs.el

```diff
@@ -70,3 +70,47 @@ Called interactively it prompts for a directory."
     (highlight-lines-matching-regexp "byebug")
     (highlight-lines-matching-regexp "binding.irb")
     (highlight-lines-matching-regexp "binding.pry")))
+
+
+;; LSP
+
+(defun spacemacs//ruby-setup-backend ()
+  "Conditionally setup ruby backend."
+  (pcase ruby-backend
+    ('lsp (spacemacs//ruby-setup-lsp))))
+
+(defun spacemacs//ruby-setup-lsp ()
+  "Setup lsp backend."
+  (if (configuration-layer/layer-used-p 'lsp)
+    (progn
+      (lsp-ruby-enable))
+    (message "`lsp' layer is not installed, please add `lsp' layer to your dotfile.")))
+
+(defun spacemacs//ruby-setup-company ()
+  "Conditionally setup company based on backend."
+  (message "%s settuing up company" ruby-backend)
```

Comment:
s/settuing/setting

Reply:
Ups forgot to remove this, it was intended for debugging.